### PR TITLE
PAS-71 API Data Structure

### PR DESCRIPTION
**Pull request by isaiahdoyle**

### CODE REVIEW

File: src/models/courseItem.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+export const courseItemSchema = new dynamoose.Schema({
+    id: String,
+    name: String,
+    weight: Number,
+    grade: [Number, fraction],
+    dueDate: Date,
+    owner: String,
+});
+
+export interface CourseItemInterface extends Document {
```
Comment (lbare) [inline src/models/courseItem.ts:14]: Should the course interface include a 'desiredGrade' field?

File: src/models/course.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            type: Set,
+            schema: [CourseItem],
+        },
+    },
+    {
+        timestamps: true,
+    }
+);
+
+export interface CourseInterface extends Document {
```
Comment (lbare) [inline src/models/course.ts:20]: Should the course interface include a 'desiredGrade' field?

Comment (isaiahdoyle) [reply src/models/course.ts:20]: That's a good question! I feel like it should, so I'll add that asap. :)

File: src/index.ts [hunk 3]
```diff
@@ -1,15 +1,23 @@
...
 app.get('/', (req, res) => {
-	res.send('Successful PassrAPI!');
+    res.send('Successful PassrAPI!');
 });
 
-app.listen(port, () => {
-	console.log('Running Passr API.');
-});
+(async () => {
+    console.log('before init');
```
Comment (BraidenCutforth) [inline src/index.ts:16]: Remove the console.log

File: src/index.ts [hunk 4]
```diff
@@ -1,15 +1,23 @@
...
+    dynamoose.aws.ddb.local();
```
Comment (BraidenCutforth) [inline src/index.ts:17]: This should be in the initDb only

File: src/initDB.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+                create: false,
+                waitForActive: {
+                    enabled: false,
+                },
+            });
+        } else {
+            dynamoose.aws.ddb.local();
+        }
+    } catch (err) {
+        console.error(err);
```
Comment (BraidenCutforth) [inline src/initDB.ts:25]: We should probably fail to start if something throws an error above. Not sure what the node equivalent of c's `exit(1)` is.

File: src/mock/index.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import Course from '../models/course';
```
Comment (BraidenCutforth) [inline src/mock/index.ts:1]: Amazing

**Review by BraidenCutforth — CHANGES_REQUESTED** [review]

**Comment by isaiahdoyle** [comment]
@BraidenCutforth Alright, I believe all required changes should be applied now :)

File: src/models/user.ts [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+        },
+    },
+    {
+        timestamps: true,
+    }
+);
+
+export interface UserInterface extends Document {
+    id: string;
+    semesters: SemesterInterface[];
```
Comment (BraidenCutforth) [inline src/models/user.ts:20]: :+1:

File: src/models/user.ts [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+import Semester, { SemesterInterface } from './semester';
+
+export const userSchema = new dynamoose.Schema(
+    {
+        id: String,
+        semesters: {
+            type: Array,
+            schema: Semester,
+        },
+    },
```
Comment (BraidenCutforth) [inline src/models/user.ts:12]: I know user Id might make more sense in terms of logic, but I think we should consider doing owner exactly as you did in the previous ones... thoughts?

File: src/routers/course.ts [hunk 9]
```diff
@@ -0,0 +1,24 @@
+import express from 'express';
```
Comment (BraidenCutforth) [inline src/routers/course.ts:1]: Is this work you actually did?

**Review by BraidenCutforth — APPROVED** [review]
**Please check out my comments before you merge!**

File: src/models/user.ts (hunk 8 above)
Comment (isaiahdoyle) [reply src/models/user.ts:12]: oooh i see - so make the primary key owner?

File: src/routers/course.ts (hunk 9 above)
Comment (isaiahdoyle) [reply src/routers/course.ts:1]: Could be a git issue? It doesn't say I modified anything on my end - this was Seiya's work

**Review by lbare — APPROVED** [review]
Nice work!

**Pull request merged**